ravenframework/SupervisedLearning/SupervisedLearning.py

```diff
@@ -69,6 +72,51 @@ class cls.
+    spec.addSub(featureSelection)
+    # Feature space transformation?
+    featureSpaceTransformation = InputData.parameterInputFactory('featureSpaceTransformation',
+                                                   descr=r"""Use dimensionality reduction technique to perform a trasformation of the training dataset
+                                                  into an uncorrelated one. The dimensionality of the problem will not be reduced but
+                                                  the data will be transformed in the transformed space. E.g if the number of features
+                                                  are 5, the method projects such features into a new uncorrelated space (still 5-dimensional).
+                                                  In case of time-dependent ROMs, all the samples are concatenated in a global 2D matrix
+                                                  (n_samples*n_timesteps,n_features) before applying the transformation and then reconstructed
+                                                  back into the original shape (before fitting the model).
+                                                   """)
+    transformationMethodType = InputTypes.makeEnumType("transformationMethod",
+                                                       "transformationMethodeType",
+                                                       ['PCA', 'KernelLinearPCA','KernelPolyPCA','KernelRbfPCA','KernelSigmoidPCA','KernelCosinePCA', 'ICA'])
+    featureSpaceTransformation.addSub(InputData.parameterInputFactory('transformationMethod',contentType=transformationMethodType,
+        descr=r"""Transformation method to use. Eight options (5 Kernel PCAs) are available:
+                  \begin{itemize}
+                    \item \textit{PCA}, Principal Component Analysis;
+                    \item \textit{KernelLinearPCA}, Kernel (Linear) Principal component analysis;
+                    \item \textit{KernelPolyPCA}, Kernel (Poly) Principal component analysis;
+                    \item \textit{KernelRbfPCA}, Kernel(Rbf) Principal component analysis;
+                    \item \textit{KernelSigmoidPCA}, Kernel (Sigmoid) Principal component analysis;
+                    \item \textit{KernelCosinePCA}, Kernel (Cosine) Principal component analysis;
+                    \item \textit{ICA}, Independent component analysis;
+                   \end{itemize}
+
+        """, default="PCA"))
+    featureSpaceTransformation.addSub(InputData.parameterInputFactory('parametersToInclude',contentType=InputTypes.StringListType,
+        descr=r"""List of IDs of features/variables to include in the transformation process.""", default=None))
+
+    spaceEnum = InputTypes.makeEnumType("spaceEnum","spaceEnumType",['Feature', 'Target'])
+    featureSpaceTransformation.addSub(InputData.parameterInputFactory('whichSpace',contentType=spaceEnum,
+        descr=r"""Which space to search? Target or Feature?""", default="Feature"))
```

Comment:
In FeatureSelectionBase, lower cases for 'Feature' and 'Target' are used, could you check the code and make it consistent?

Reply:
done